### PR TITLE
ADD: New sniff for final controller

### PR DESCRIPTION
**Pull request by mantebalzaraviciute**
Issue: https://github.com/JumpTwentyFour/php-coding-standards/issues/25

### Added a new sniff for checking that controllers are final
- This also works with fixer as well. ✌🏻

Example:
Screenshot 2023-03-23 at 12 13 21

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -31,6 +31,11 @@
         "symplify/easy-coding-standard": "^10.0.6",
         "nikic/php-parser": "^4.0"
     },
+    "autoload": {
+        "psr-4": {
+            "PhpCodingStandards\\": "src"
```
Comment (mmorganJ24) [inline composer.json:36]: Maybe this should be `Jump24\\PhpCodingStandards\\` just to be really clear when importing bits that it's from this package and not another?

Comment (mantebalzaraviciute) [reply composer.json]: I made it JumpTwentyFour to match the laravel coding standards repo :)

File: src/Sniffs/FinalControllerSniff.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        $tokens = $phpcsFile->getTokens();
+
+        $className = $phpcsFile->getDeclarationName($classPointer);
+
+        if (str_contains($className, 'Test')) {
+            return;
+        }
+
+        if (str_contains($className, 'Controller') === false) {
```
Comment (mmorganJ24) [inline src/Sniffs/FinalControllerSniff.php:34]: Will this kick off about base Laravel controllers that extend the framework's one?

Comment (AlexJump24) [reply src/Sniffs/FinalControllerSniff.php:34]: Good point, this should probably exclude the default `Controller.php`.

Comment (mantebalzaraviciute) [reply src/Sniffs/FinalControllerSniff.php:34]: I told Matt his and he sneakily put it in a comment as his idea :D

Comment (mmorganJ24) [reply src/Sniffs/FinalControllerSniff.php:34]: You told me to put it on here 😆

Comment (mantebalzaraviciute) [reply src/Sniffs/FinalControllerSniff.php:38]: @AlexJump24 @mmorganJ24 I added an early return for the rule if the classname is just controller return. But we can also add excludes in the ecs.php config when we need to as well

**Pull request merged**